src/runtime/arc.ts

```diff
@@ -509,27 +511,32 @@ ${this.activeRecipe.toString()}`;
     return arc;
   }
 
-  /** in-flight instantiate, or Promise.resolve() if not executing */
-  private nextInstantiate: Promise<void> = Promise.resolve();
-
   /**
-   * Adds a new recipe to instantiate in the Arc.
+   * Instantiates the given recipe in the Arc.
    *
    * Executes the following steps:
-   * - TBD1
-   * - TBD2
+   *
+   * - Merges the recipe into the Active Recipe
+   * - Populates missing slots.
+   * - Processes the Handles and creates stores for them.
+   * - Instantiates the new Particles
+   * - Passes these particles for initialization in the PEC
+   * - For non-speculative Arcs processes instantiatePlanCallbacks
    *
    * Waits for completion of an existing Instantiate before returning.
    */
   async instantiate(recipe: Recipe): Promise<void> {
+
     assert(recipe.isResolved(), `Cannot instantiate an unresolved recipe: ${recipe.toString({showUnresolved: true})}`);
     assert(recipe.isCompatible(this.modality),
       `Cannot instantiate recipe ${recipe.toString()} with [${recipe.modality.names}] modalities in '${this.modality.names}' arc`);
 
-    await this.nextInstantiate;
-    this.nextInstantiate = this._doInstantiate(recipe);
-    await this.nextInstantiate;
-    this.nextInstantiate = Promise.resolve();
+    const release = await this.instantiateMutex.acquire();
```

Comment:
Why not bake the knowledge about `release` into the mutex code (so the caller only need supply a function)?

Reply:
Example?  I think you need to have an easy way to control how you release and this plays nice with try-finally pattern.

A java style [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) api would be kind of interesting.  Can we execute code when a variable drops from scope?